src/System.Data.SqlClient/src/System/Data/Common/DbEnumerator.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Data;
+using System.Data.ProviderBase;
+using System.Diagnostics;
+using System.ComponentModel;
+
+namespace System.Data.Common
+{
+    public class DbEnumerator : IEnumerator
+    {
+        internal DbDataReader _reader;
+        internal DbDataRecord _current;
+        internal SchemaInfo[] _schemaInfo; // shared schema info among all the data records
+        private FieldNameLookup _fieldNameLookup;
+
+        private bool _closeReader;
+
+        public DbEnumerator(DbDataReader reader, bool closeReader)
+        {
+            if (null == reader)
+            {
+                throw ADP.ArgumentNull("reader");
+            }
+            _reader = reader;
+            _closeReader = closeReader;
+        }
+
+
+        public object Current
+        {
+            get
+            {
+                return _current;
+            }
+        }
+
+        /*public IDataRecord Current {
+            get {
+                return _current;
+            }
+        }*/
+
+        /*
+                virtual internal IDataRecord NewRecord(SchemaInfo[] si, object[] values, PropertyDescriptorCollection descriptors) {
+                    return new DbDataRecord(si, values, descriptors);
+                }
+
+                virtual internal void GetValues(object[] values) {
+                    _reader.GetValues(values);
+                }
+        */
```

Comment:
In general it'd be nice to delete commented out code, or if there's a reason to keep it, add a comment for why, ideally including an associated issue number for when it'll be brought back.